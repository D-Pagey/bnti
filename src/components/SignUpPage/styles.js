import styled from 'styled-components';
import { margin } from 'polished';
import { fontWeight, spacing } from '../../tokens';

export const Wrapper = styled.div`
    ${margin(spacing.huge, 'auto', spacing.none)};
    width: 30%;
`;

export const Title = styled.h1`
    text-align: center;
`;

export const Form = styled.form`
    display: flex;
    flex-direction: column;
    width: 300px;
`;

export const Label = styled.label`
    align-items: center;
    display: flex;
    font-weight: ${fontWeight.medium};
    justify-content: space-between;
    margin: 5px 0;
`;
