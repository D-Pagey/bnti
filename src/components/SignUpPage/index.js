import React, { useState, useContext } from 'react';
import AuthContext from '../../contexts/auth';
import * as S from './styles';

const SignUp = () => {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [username, setUsername] = useState('');
    const { authSignUp } = useContext(AuthContext);

    const handleChange = (handler) => ({ target: { value } }) => handler(value);
    const validateForm = () => email.length && password.length && username.length;

    const handleSubmit = async (event) => {
        event.preventDefault();
        authSignUp({ email, password });
    };

    return (
        <S.Wrapper>
            <S.Title>Sign Up</S.Title>
            <S.Form onSubmit={handleSubmit}>
                <S.Label htmlFor="username">
                    Username:
                    <input type="text" id="username" value={username} onChange={handleChange(setUsername)} />
                </S.Label>
                <S.Label htmlFor="email">
                    Email:
                    <input type="text" id="name" value={email} onChange={handleChange(setEmail)} />
                </S.Label>
                <S.Label htmlFor="password">
                    Password:
                    <input
                        type="password"
                        id="password"
                        value={password}
                        onChange={handleChange(setPassword)}
                        placeholder="FoobarTest1234!"
                    />
                </S.Label>
                <button type="submit" disabled={!validateForm()}>
                    Sign Up
                </button>
            </S.Form>
        </S.Wrapper>
    );
};

export default SignUp;
