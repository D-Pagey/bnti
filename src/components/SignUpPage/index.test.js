import React from 'react';
import SignUpPage from '.';

describe('SignUpPage component', () => {
    it('should render', () => {
        const { container } = render(<SignUpPage />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
