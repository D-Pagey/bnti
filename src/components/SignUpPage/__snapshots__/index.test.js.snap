// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage component should render 1`] = `
.c0 {
  margin-top: 3rem;
  margin-right: auto;
  margin-bottom: 0rem;
  margin-left: auto;
  width: 30%;
}

.c1 {
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 300px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 500;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 5px 0;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Sign Up
  </h1>
  <form
    class="c2"
  >
    <label
      class="c3"
      for="username"
    >
      Username:
      <input
        id="username"
        type="text"
        value=""
      />
    </label>
    <label
      class="c3"
      for="email"
    >
      Email:
      <input
        id="name"
        type="text"
        value=""
      />
    </label>
    <label
      class="c3"
      for="password"
    >
      Password:
      <input
        id="password"
        placeholder="FoobarTest1234!"
        type="password"
        value=""
      />
    </label>
    <button
      disabled=""
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;
